source/blog/2017-01-01-ovirt-system-tests-to-the-rescue.html.md

```diff
@@ -0,0 +1,63 @@
+---
+title: oVirt System Tests to the Rescue!&mdash;How to run end-to-end oVirt tests on your patch.
+author: eedri
+tags: infrastructure, developers, ci
+date: 2017-01-01 10:00:00 CET
+---
+
+Today, when an oVirt developer pushes a patch to review on oVirt Gerrit, various validations are triggered in CI via the ['check-patch'](http://ovirt-infra-docs.readthedocs.io/en/latest/CI/Build_and_test_standards.html) job, as defined by the project maintainers.
+Usually these jobs includes 'unit-tests', 'db tests', static analysis checks and even an occasional 'functional test'.
+While it might seem that it covers alot and gives a good indication that the patch is good to be merged, unfourtunately it is not always the case.
+
+The reason it's not enough lies in oVirt's complexity and the fact its a Virtualization project, which means the only real way to know if your patch didn't break things is to
+install oVirt and try running a few basic commands, like 'adding host', 'adding vm', 'creating snapshots' and other tasks you can only do if you have
+a full oVirt system up and running. Here is where OST comes in!
+ 
+## oVirt System Tests
+
+[oVirt system tests](http://ovirt-system-tests.readthedocs.io) is a testing framework written in Python, using 'python-nose' and oVirt Python SDK and runs on auto-generated VMs created by [Lago](http://lago.readthedocs.io).
+It is used by the oVirt CI to run post merge end-to-end testing which runs on a fully deployed oVirt envrionment and has been proven to detect multiple
+regressions so far on merged commits from oVirt projects.
+
+## The Current Status ( and why it is never enough )
+
+So you may ask yourself - If we have OST running after (almost) every merged commit, where is the problem?
+You might have guessed the answer which is already in the question - it's only 'AFTER', which means it's detected too late in the development cycle.
+The result of such breakage in CI means that any other developer using the same branch will now be blocked from working / verifying his patch
+because existing HEAD is broken, and usually it takes some time to either revert the offending patch or send a fix ( and that's after the relevant
+people were found and started debugging the issue ). On some occasions in the past, it even took a few days to see a fix merged.
+During that time, develpoers and testers were blocked, not an ideal status.<br>
```

Comment:
They may be. Leave, and I will double check when running the final rendering check.